### PR TITLE
add perf_event to cgroups v1

### PR DESCRIPTION
**Pull request by fbrv**

### CODE REVIEW

File: src/cgroups/v1/perf_event.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+use crate::cgroups::v1::Controller;
+use oci_spec::{LinuxResources};
+use std::path::Path;
+
+pub struct PerfEvent {}
+
+impl Controller for PerfEvent {
+    type Resource = ();
+
+    fn apply(_linux_resources: &LinuxResources, _cgroup_root: &Path) -> anyhow::Result<()> {
```
Comment (Furisto) [inline src/cgroups/v1/perf_event.rs:11]: Please replace with
```suggestion
     use anyhow::Result;

    fn apply(_linux_resources: &LinuxResources, _cgroup_root: &Path) -> Result<()> {
```

File: src/cgroups/v1/perf_event.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    use std::fs;
+
+    use nix::unistd::Pid;
+
+    use super::*;
+    use crate::cgroups::{common::CGROUP_PROCS, test::setup};
+
+    #[test]
+    fn test_add_task() {
+        let (tmp, procs) = setup("test_perf_event_apply", CGROUP_PROCS);
```
Comment (Furisto) [inline src/cgroups/v1/perf_event.rs:31]:
```suggestion
        let (tmp, procs) = setup("test_perf_event_add_task", CGROUP_PROCS);
```

**Review by Furisto — CHANGES_REQUESTED** [review]

File: src/cgroups/v1/perf_event.rs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+
+use crate::cgroups::v1::Controller;
+use oci_spec::{LinuxResources};
+use std::path::Path;
+
```
Comment (Furisto) [inline src/cgroups/v1/perf_event.rs:5]:
```suggestion
 use anyhow::Result;
```

**Comment by Furisto** [comment]
@fbrv Sorry for the confusion. With that suggestion I meant that you should add the use "above", not literally in the trait implementation.

**Comment by fbrv** [comment]
> @fbrv Sorry for the confusion. With that suggestion I meant that you should add the use "above", not literally in the trait implementation.

My fault, I haven't properly checked the suggestion, still learning rust :) hopefully in the next contributions will not happens anymore

**Comment by Furisto** [comment]
@fbrv You need to run cargo fmt, otherwise the check will not succeed.

**Comment by fbrv** [comment]
> @fbrv You need to run cargo fmt, otherwise the check will not succeed.

Run multiple times but for some reason it did not catch any error. Reinstalled rust and now seems working

**Review by Furisto — APPROVED** [review]

**Comment by Furisto** [comment]
🎉

**Pull request merged**